pkg/controller/perconaservermongodb/pbm.go

```diff
@@ -188,6 +189,50 @@ func isResyncNeeded(currentCfg *config.Config, newCfg *config.Config) bool {
 	return false
 }
 
+func (r *ReconcilePerconaServerMongoDB) reconcilePiTRStorageLegacy(
+	ctx context.Context,
+	pbm backup.PBM,
+	cr *psmdbv1.PerconaServerMongoDB,
+) error {
+	log := logf.FromContext(ctx)
+	// if PiTR is enabled user can configure only one storage
+	var storage psmdbv1.BackupStorageSpec
+	for name, stg := range cr.Spec.Backup.Storages {
+		storage = stg
+		log.Info("Configuring PBM with storage", "storage", name)
+		break
+	}
+
+	var secretName string
+	switch storage.Type {
+	case psmdbv1.BackupStorageS3:
+		secretName = storage.S3.CredentialsSecret
+	case psmdbv1.BackupStorageAzure:
+		secretName = storage.Azure.CredentialsSecret
+	}
+
+	if secretName != "" {
+		exists, err := secretExists(ctx, r.client, types.NamespacedName{Name: secretName, Namespace: cr.Namespace})
+		if err != nil {
+			return errors.Wrap(err, "check storage credentials secret")
+		}
+
+		if !exists {
+			log.Error(nil, "Storage credentials secret does not exist", "secret", secretName)
+			return nil
+		}
+	}
+
+	err := pbm.GetNSetConfigLegacy(ctx, r.client, cr, storage)
```

Comment:
Still wondering why this function is called `GetNSet...` since it is essentially setting. If internally it gets, sets, appends, etc is not something we should leak.

Reply:
yeah the reason is it generates the PBM config according to values in cr.yaml and sets it